src/Core/Framework/DataAbstractionLayer/Search/Criteria.php

```diff
@@ -467,10 +473,15 @@ public function setTerm(?string $term): self
     }
 
     /**
-     * @param string[]|array<int, string[]> $ids
+     * @param string[]|array<int, string[]>|null $ids
      */
-    public function cloneForRead(array $ids = []): Criteria
+    public function cloneForRead(?array $ids = null): Criteria
     {
+        // To make behaviour consistent with constructor
+        if ($ids === []) {
+            $ids = null;
```

Comment:
I think we should also throw InconsistentCriteriaIdsException here

Reply:
I'd need to change the `EntityRepository` so an empty array of ids is being catched there during searching, but that should be possible.

Reply:
Done!